code/core/src/manager/components/preview/tools/share.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { useMemo, useState } from 'react';
+
+import {
+  IconButton,
+  TooltipLinkList,
+  WithTooltip,
+  getStoryHref,
+} from 'storybook/internal/components';
+import type { Addon_BaseType } from 'storybook/internal/types';
+
+import { global } from '@storybook/global';
+import { BugIcon, LinkIcon, ShareIcon } from '@storybook/icons';
+
+import copy from 'copy-to-clipboard';
+import { QRCodeSVG as QRCode } from 'qrcode.react';
+import { Consumer, types, useStorybookApi } from 'storybook/manager-api';
+import type { Combo } from 'storybook/manager-api';
+import { styled, useTheme } from 'storybook/theming';
+
+import { Shortcut } from '../../../container/Menu';
+
+const { PREVIEW_URL, document } = global as any;
+
+const mapper = ({ state }: Combo) => {
+  const { storyId, refId, refs } = state;
+  const { location } = document;
+  // @ts-expect-error (non strict)
+  const ref = refs[refId];
+  let baseUrl = `${location.origin}${location.pathname}`;
+
+  if (!baseUrl.endsWith('/')) {
+    baseUrl += '/';
+  }
+
+  return {
+    refId,
+    baseUrl: ref ? `${ref.url}/iframe.html` : (PREVIEW_URL as string) || `${baseUrl}iframe.html`,
+    storyId,
+    queryParams: state.customQueryParams,
+  };
+};
+
+const QRContainer = styled.div(() => ({
+  display: 'flex',
+  alignItems: 'center',
+  padding: 8,
+  maxWidth: 200,
+}));
+
+const QRImageContainer = styled.div(() => ({
+  width: 64,
+  height: 64,
+  marginRight: 12,
+  backgroundColor: 'white',
+  padding: 2,
+}));
+
+const QRImage = ({ value }: { value: string }) => {
+  const theme = useTheme();
+  return (
+    value && (
+      <QRImageContainer>
+        <QRCode value={value} marginSize={0} size={60} fgColor={theme.color.darkest} />
+      </QRImageContainer>
+    )
+  );
+};
```

Comment:
This seems odd to me.

According to types `value` is always a `string`, but then you check if it's `falsy`?